### ISSUE TITLE
pausing and resuming leaves focus for keyboard nav

### ISSUE THREAD

**Issue by BryceAG** [phetsims/friction#52]
Pausing and resuming leaves focus from the sim, and after it is resumed the person must tab to get keyboard nav back up, but it goes to the next object. This could be disorienting for the person using keyboard nav, as they did nothing to leave focus. To make sure it wasn't just me clicking the resume button that caused the loss of focus I used pause in 5 seconds and resume in 5 seconds:
![ezgif com-video-to-gif 1](https://user-images.githubusercontent.com/15327201/33679209-59c80ef4-da7b-11e7-82f9-4edff6544649.gif)

Is this intended?

related to phetsims/QA#75

**Comment by zepumph** [comment]
@jessegreenberg is the lol guy.

**Comment by jessegreenberg** [comment]
Still a problem when I remove
```
 this.accessibleDOMElement.hidden = !this._interactive;
```
from Display. I wonder if this is happening in the active wrapper.

**Comment by BryceAG** [comment]
Update: this is what happens in Chrome. In Firefox, the object stays highlighted as though it is selected, but does not respond to keyboard nav. It will continue to do the action it was doing when paused (so if it was being dragged left, after resuming it will go all the way to the left) and Tab must pressed many times before returning to keyboard nav for the sim

**Comment by jessegreenberg** [comment]
Thanks @BryceAG.

I added this to active.html:
```js
  setInterval( function() {
    simIFrameClient.invoke( sim.camelCaseName + '.activeProperty', 'setValue', [ !lastActiveValue ] );
  }, 5000 )
```

And it is still a problem in Chrome when the line in https://github.com/phetsims/friction/issues/52#issuecomment-349749754 is added, but bug is fixed when that line is removed.

**Comment by BryceAG** [comment]
Also in Edge it doesn't load for LOL

**Comment by jessegreenberg** [comment]
> Tab must pressed many times before returning to keyboard nav for the sim

Verified. It looks like focus gets stuck in the iframe. The doesn't happen in "Friction iframe test".

**Comment by jessegreenberg** [comment]
Something like this could be added to Display.js to resetore focus:
```js
        if ( this._interactive && this._activeNode ) {
          this._activeNode.focus();
          this._activeNode = null;
        }
        else {
          this._activeNode = Display.focusedNode;
        }
```

**Comment by jessegreenberg** [comment]
I have questions about whether or not this is desirable, and we should discuss with @terracoda and @emily-phet before continuing.  The typical use case for "Pause" is to get the student to stop playing with the sim and listen to the teacher or go to other content. If while the sim is paused, the user navigates to something else with the keyboard it could be frustrating to interrupt what they were doing and force focus back into the sim. This feels bad for a11y.  On the other hand, it is slightly inconvenient that user needs to navigate from the top of the sim once it is made active again.

I believe this can be deferred until we discuss this in greater detail in https://github.com/phetsims/a11y-research/issues/69. @emily-phet @terracoda do you agree?

**Comment by terracoda** [comment]
It would be good to discuss this. 
Pressing the pause button would naturally move keyboard focus out of the sim. When returning to the sim, pressing the tab key would naturally move focus to next item.

I think the best we can do - when returning to the sim - is to try to put them on the last item that had focus. This would be akin using ESC key to close a dialog. I'm not sure there is an easy way to do that. Play with a fixed Tab index might be dangerous. Is there a way to record which item had focus before the pause button was focused and pressed?

**Comment by jessegreenberg** [comment]
> Is there a way to record which item had focus before the pause button was focused and pressed?

Yes, we can do that.

**Comment by jessegreenberg** [comment]
https://github.com/phetsims/friction/issues/52#issuecomment-349761547

I forgot about this browser specific bug which creates a focus trap so I don't think this can be deferred.

**Comment by jessegreenberg** [comment]
The above commit fixes the focus trap in FF and makes the behavior like @terracoda suggested. When returning to the sim, focus will return to  the last item that had focus instead of returning to the next one.

**Comment by emily-phet** [comment]
@jessegreenberg @terracoda If I'm understanding correctly, it seems like this issue has been addressed? The solution of having focus return to the place it was after pausing seems like a good decision to me. Let me know if there are any questions for me.

**Comment by terracoda** [comment]
@emily-phet, I too, believe this issue has been resolved.

**Comment by jessegreenberg** [comment]
Thanks @emily-phet and @terracoda, this should be addressed. Will be verified in the next round of testing.

**Comment by zepumph** [comment]
Cherry-picked into scenery's friction-1.4 branch

**Comment by phet-steele** [comment]
Works well for me.
